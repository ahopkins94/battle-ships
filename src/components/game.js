import React from 'react'

export class Game extends React.Component {

  render() {
    return(
      <div id="Game">
      <div>
        Hello
      </div>
      </div>
    )
  }
}
