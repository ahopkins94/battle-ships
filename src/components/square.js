import React from 'react'

export function Square(props) {
  return(
      <button className="square">
      </button>
  )
}
