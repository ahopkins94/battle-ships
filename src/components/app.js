import React from 'react'
import { Switch, Route } from 'react-router-dom'
import { BoatLocationForm } from './boatLocationForm'
import { Game } from './game'

export class App extends React.Component {

  render() {
    return(
      <main>
      <Switch>
        <Route exact path='/' component={BoatLocationForm} />
        <Route exact path='/game' component={Game} />
      </Switch>
      </main>
    )
  }
}
