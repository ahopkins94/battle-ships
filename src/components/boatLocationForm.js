import React from 'react'
import { Game } from './game'

export class BoatLocationForm extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      dinghyLocation1: null,
      dinghyLocation2: null,
      dinghyLocation3: null,
      submarineLocation1: null,
      submarineLocation2: null,
      submarineLocation3: null,
      submarineLocation4: null,
      destroyerLocation1: null,
      destroyerLocation2: null,
      destroyerLocation3: null,
      destroyerLocation4: null,
      destroyerLocation5: null,
      destroyerLocation6: null,
      battleshipLocation1: null,
      battleshipLocation2: null,
      battleshipLocation3: null,
      battleshipLocation4: null,
      battleshipLocation1: null,
      battleshipLocation5: null,
      battleshipLocation6: null,
      battleshipLocation7: null,
      battleshipLocation8: null,
      allPlayerLocations: Array(61).fill(null),
      submitted: false
    }

  }

  handleChange(event) {
    this.setState({[event.target.name]: event.target.value})
    console.log(this.state)
  }

  handleSubmit(event) {
    var locations = [this.state.dinghyLocation1, this.state.dinghyLocation2, this.state.dinghyLocation3, this.state.submarineLocation1, this.state.submarineLocation2, this.state.submarineLocation3, this.state.submarineLocation4, this.state.destroyerLocation1, this.state.destroyerLocation2, this.state.destroyerLocation3, this.state.destroyerLocation4, this.state.destroyerLocation5, this.state.destroyerLocation6, this.state.battleshipLocation1, this.state.battleshipLocation2, this.state.battleshipLocation3, this.state.battleshipLocation4, this.state.battleshipLocation5, this.state.battleshipLocation6, this.state.battleshipLocation7, this.state.battleshipLocation8]
    let clonedAllPlayerLocations = this.state.allPlayerLocations
    var i
    for (i = 0; i < locations.length; i++) {
      var location = locations[i]
      clonedAllPlayerLocations[location] = "X"
    }
    this.setState({allPlayerLocations: clonedAllPlayerLocations})
    event.preventDefault()
    this.setState({
      submitted: true
    })
  }

  render() {
    if (this.state.submitted === true) {
      return(
        <Game allPlayerLocations={this.state.allPlayerLocations} />
      )
    }
    return(
      <div id="boatLocationForm">
      <div>Please enter your boats' coordinates below:</div>
      <div>
      <form onSubmit={(e) => this.handleSubmit(e)}>
        The Dinghy:
        <br></br><input type="text" name="dinghyLocation1" placeholder="Coordinate 1" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="text" name="dinghyLocation2" placeholder="Coordinate 2" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="text" name="dinghyLocation3" placeholder="Coordinate 3" onChange={(e) => this.handleChange(e, "dinghyLocation3")}></input>
        <br></br>The Submarine:
        <br></br><input type="text" name="submarineLocation1" placeholder="Coordinate 1" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="text" name="submarineLocation2" placeholder="Coordinate 2" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="text" name="submarineLocation3" placeholder="Coordinate 3" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="text" name="submarineLocation4" placeholder="Coordinate 4" onChange={(e) => this.handleChange(e)}></input>
        <br></br>The Destroyer:
        <br></br><input type="text" name="destroyerLocation1" placeholder="Coordinate 1" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="text" name="destroyerLocation2" placeholder="Coordinate 2" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="text" name="destroyerLocation3" placeholder="Coordinate 3" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="text" name="destroyerLocation4" placeholder="Coordinate 4" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="text" name="destroyerLocation5" placeholder="Coordinate 5" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="text" name="destroyerLocation6" placeholder="Coordinate 6" onChange={(e) => this.handleChange(e)}></input>
        <br></br>The Battleship:
        <br></br><input type="text" name="battleshipLocation1" placeholder="Coordinate 1" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="text" name="battleshipLocation2" placeholder="Coordinate 2" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="text" name="battleshipLocation3" placeholder="Coordinate 3" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="text" name="battleshipLocation4" placeholder="Coordinate 4" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="text" name="battleshipLocation5" placeholder="Coordinate 5" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="text" name="battleshipLocation6" placeholder="Coordinate 6" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="text" name="battleshipLocation7" placeholder="Coordinate 7" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="text" name="battleshipLocation8" placeholder="Coordinate 8" onChange={(e) => this.handleChange(e)}></input>
        <br></br><input type="submit" value="submit"></input>
      </form>
      </div>
      </div>
    )
  }
}
