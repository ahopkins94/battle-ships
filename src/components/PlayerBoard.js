import React from 'react'
import { Square } from './square'

export class PlayerBoard extends React.Component {

  formatSquare() {
    var squares = [];
    for (var i = 0; i < 100; i++) {
        squares.push(<Square key={i} />);
    }
    return(
     <div id="squaresClass">
       <div className="board-row">{squares.slice(0, 10)}</div>
       <div className="board-row">{squares.slice(10, 20)}</div>
       <div className="board-row">{squares.slice(20, 30)}</div>
       <div className="board-row">{squares.slice(30, 40)}</div>
       <div className="board-row">{squares.slice(40, 50)}</div>
       <div className="board-row">{squares.slice(50, 60)}</div>
       <div className="board-row">{squares.slice(60, 70)}</div>
       <div className="board-row">{squares.slice(70, 80)}</div>
       <div className="board-row">{squares.slice(80, 90)}</div>
       <div className="board-row">{squares.slice(90, 100)}</div>
     </div>
     )
  }

  render() {
    return (
      <div>
      {this.formatSquare()}
      </div>
    )
  }
}
