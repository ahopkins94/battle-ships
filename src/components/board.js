import React from 'react'
import { Square } from './square'

export class Board extends React.Component {

  formatSquare() {
    var squares = [];
    for (var i = 0; i < 100; i++) {
        squares.push(<Square key={i} />);
    }
    return(
     <div id="squaresClass">
       <div>{squares.slice(0, 10)}</div>
       <div>{squares.slice(10, 20)}</div>
       <div>{squares.slice(20, 30)}</div>
       <div>{squares.slice(30, 40)}</div>
       <div>{squares.slice(40, 50)}</div>
       <div>{squares.slice(50, 60)}</div>
       <div>{squares.slice(60, 70)}</div>
       <div>{squares.slice(70, 80)}</div>
       <div>{squares.slice(80, 90)}</div>
       <div>{squares.slice(90, 100)}</div>
     </div>
     )
  }

  render() {
    return (
      <div>
      {this.formatSquare()}
      </div>
    )
  }
}
