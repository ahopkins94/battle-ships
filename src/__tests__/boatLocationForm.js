import React from 'react'
import { BoatLocationForm } from '../components/boatLocationForm'
import { shallow, mount, render } from 'enzyme'

describe('BoatLocationForm', function() {
  var wrapper

  it("should match the snapshot", function() {
    wrapper = shallow(<BoatLocationForm />)
    expect(wrapper).toMatchSnapshot()
  })

  it("should have 3 divs", function() {
    wrapper = shallow(<BoatLocationForm />)
    const divs = wrapper.find('div')
    expect(divs.length).toBe(3)
  })

  it("should set state and change dinghyLocation", function() {
    wrapper = mount(<BoatLocationForm />)
    wrapper.find('input').first().simulate('change', { target: { value: 'Hello' } })
    setTimeout(() => {
      expect(wrapper.state('dinghyLocation')).toEqual('Hello')
    }, 20)
  })
})
