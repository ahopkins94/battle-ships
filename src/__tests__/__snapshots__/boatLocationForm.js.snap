// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoatLocationForm should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BoatLocationForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          Please enter your boats' coordinates below:
        </div>,
        <div>
          <form>
            The Dinghy:
            <br />
            <input
              onChange={[Function]}
              placeholder="Coordinate 1"
              type="text"
              value="dinghyLocation"
            />
            <br />
            <input
              type="submit"
              value="submit"
            />
          </form>
        </div>,
      ],
      "id": "boatLocationForm",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please enter your boats' coordinates below:",
        },
        "ref": null,
        "rendered": "Please enter your boats' coordinates below:",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form>
            The Dinghy:
            <br />
            <input
              onChange={[Function]}
              placeholder="Coordinate 1"
              type="text"
              value="dinghyLocation"
            />
            <br />
            <input
              type="submit"
              value="submit"
            />
          </form>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "The Dinghy:",
              <br />,
              <input
                onChange={[Function]}
                placeholder="Coordinate 1"
                type="text"
                value="dinghyLocation"
              />,
              <br />,
              <input
                type="submit"
                value="submit"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "The Dinghy:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Coordinate 1",
                "type": "text",
                "value": "dinghyLocation",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "submit",
                "value": "submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            Please enter your boats' coordinates below:
          </div>,
          <div>
            <form>
              The Dinghy:
              <br />
              <input
                onChange={[Function]}
                placeholder="Coordinate 1"
                type="text"
                value="dinghyLocation"
              />
              <br />
              <input
                type="submit"
                value="submit"
              />
            </form>
          </div>,
        ],
        "id": "boatLocationForm",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Please enter your boats' coordinates below:",
          },
          "ref": null,
          "rendered": "Please enter your boats' coordinates below:",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form>
              The Dinghy:
              <br />
              <input
                onChange={[Function]}
                placeholder="Coordinate 1"
                type="text"
                value="dinghyLocation"
              />
              <br />
              <input
                type="submit"
                value="submit"
              />
            </form>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "The Dinghy:",
                <br />,
                <input
                  onChange={[Function]}
                  placeholder="Coordinate 1"
                  type="text"
                  value="dinghyLocation"
                />,
                <br />,
                <input
                  type="submit"
                  value="submit"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "The Dinghy:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "onChange": [Function],
                  "placeholder": "Coordinate 1",
                  "type": "text",
                  "value": "dinghyLocation",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "type": "submit",
                  "value": "submit",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "form",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
